### PR TITLE
Fix(Automate): CNX-8610 automate sdk warnings

### PR DESCRIPTION
**Pull request by oguzhankoral**
Fixing all warnings on Automate.SDK and it's test project. And added README for test project.

### CODE REVIEW

File: Automate/Tests/Speckle.Automate.Sdk.Tests.Integration/GetAutomationStatus.cs [hunk 1]
```diff
@@ -3,34 +3,34 @@
...
+public struct ProjectAutomationStatus
 {
   public ModelAutomationStatus Model { get; set; }
 }
 
-public class AutomationStatusResponseModel
+public struct AutomationStatusResponseModel
 {
   public ProjectAutomationStatus Project { get; set; }
 }
```
Comment (AlanRynne) [inline Automate/Tests/Speckle.Automate.Sdk.Tests.Integration/GetAutomationStatus.cs:36]: Out of scope of this PR but this seems rather odd to have in the Integration tests instead of the SDK project 🤔

File: Automate/Speckle.Automate.Sdk/Schema/AutomationRunData.cs [hunk 2]
```diff
@@ -1,8 +1,7 @@
-# nullable enable
 namespace Speckle.Automate.Sdk.Schema;
 
 ///<summary>
-///Values of the project, model and automation that triggere this function run.
+///Values of the project, model and automation that trigger this function run.
```
Comment (AlanRynne) [inline Automate/Speckle.Automate.Sdk/Schema/AutomationRunData.cs:4]: I think the right tense would be `triggered` in this case

File: Automate/Speckle.Automate.Sdk/Runner.cs [hunk 3]
```diff
@@ -85,7 +86,7 @@ public static async Task<int> Main(string[] args, Func<AutomationContext, Task>
   }
 
   /// <summary>
-  /// Main entrypoint to execute an Automate function with input data of type <see cref="TInput"/>
+  /// Main entrypoint to execute an Automate function with input data of type.
```
Comment (AlanRynne) [inline Automate/Speckle.Automate.Sdk/Runner.cs:89]: Why was this deleted? cref references to types are usually better as they allow to navigate to them on the xml docs

File: Automate/Speckle.Automate.Sdk/AutomationContext.cs [hunk 4]
```diff
@@ -137,7 +158,7 @@ public void SetContextView(List<string>? resourceIds = null, bool includeSourceM
 
     if (linkResources.Count == 0)
     {
-      throw new Exception("We do not have enough resource ids to compose a context view");
+      throw new SpeckleException("We do not have enough resource ids to compose a context view");
```
Comment (AlanRynne) [inline Automate/Speckle.Automate.Sdk/AutomationContext.cs:161]: Maybe a more specific exception would be desirable here?

File: Automate/Speckle.Automate.Sdk/AutomationContext.cs [hunk 5]
```diff
@@ -15,40 +15,41 @@ public class AutomationContext
...
-      serverTransport = serverTransport,
-      speckleToken = speckleToken,
-      memoryTransport = new MemoryTransport(),
-      initTime = initTime,
+      _serverTransport = serverTransport,
+      _speckleToken = speckleToken,
+      _memoryTransport = new MemoryTransport(),
+      _initTime = initTime,
       AutomationResult = new AutomationResult(),
     };
```
Comment (AlanRynne) [inline Automate/Speckle.Automate.Sdk/AutomationContext.cs:54]: This should have a constructor for it. Out of scope of this PR, we should open a ticket for this.

**Review by AlanRynne — CHANGES_REQUESTED** [review]
Super nice overall! `O warnings` 🙌🏼 Just have some comments on a couple things

File: Automate/Speckle.Automate.Sdk/Runner.cs (hunk 3 above)
Comment (oguzhankoral) [reply Automate/Speckle.Automate.Sdk/Runner.cs:89]: The warning was pushing me I believe 🤔

Comment (JR-Morgan) [reply Automate/Speckle.Automate.Sdk/Runner.cs:89]: I think the correct tag here would be `<TypeParamRef cref="TInput/>`

File: Automate/Speckle.Automate.Sdk/Schema/AutomationRunData.cs (hunk 2 above)
Comment (oguzhankoral) [reply Automate/Speckle.Automate.Sdk/Schema/AutomationRunData.cs]: resolved

File: Automate/Speckle.Automate.Sdk/AutomationContext.cs (hunk 4 above)
Comment (oguzhankoral) [reply Automate/Speckle.Automate.Sdk/AutomationContext.cs:161]: how about ?

```c#
public class SpeckleContextViewException : SpeckleException { }
```

File: Automate/Speckle.Automate.Sdk/Runner.cs (hunk 3 above)
Comment (oguzhankoral) [reply Automate/Speckle.Automate.Sdk/Runner.cs:89]: It is already there at line 93

Comment (JR-Morgan) [reply Automate/Speckle.Automate.Sdk/Runner.cs:89]: no that's the `typeparam` tag. this would be a reference to the name of that type param
If you look at the rendered doc panel, you'll see the indented summary tag would read "...input data of type TInput"

image

Comment (JR-Morgan) [reply Automate/Speckle.Automate.Sdk/Runner.cs:89]: I can walk you through some of the XML doc tags after standup if you'd like, there's several lesser know ones like `typeparamref` that come in handy when writing XML docs

Comment (oguzhankoral) [reply Automate/Speckle.Automate.Sdk/Runner.cs:89]: gotcha, yup there are different. I'm adding it as in your screenshot, then we can chat about it after SU

**Review by JR-Morgan — CHANGES_REQUESTED** [review]
One last change please. Can you uncomment these lines in the automate csproj

https://github.com/specklesystems/speckle-sharp/blob/4978c10f9852f52043b35957f4e3bf4d9e60ce0f/Automate/Speckle.Automate.Sdk/Speckle.Automate.Sdk.csproj#L15-L17

And also copy these lines to the automate test project.

These will make any warnings fail in CI, so there's no chance of any warnings being shipped now once this change is made!

File: Automate/Tests/Speckle.Automate.Sdk.Tests.Integration/Speckle.Automate.Sdk.Tests.Integration.csproj [hunk 6]
```diff
@@ -10,6 +10,10 @@
         <NoWarn>$(NoWarn);CA2007</NoWarn>
```
Comment (JR-Morgan) [inline Automate/Tests/Speckle.Automate.Sdk.Tests.Integration/Speckle.Automate.Sdk.Tests.Integration.csproj:10]: One last quick change (Sorry this one is on me, I forgot to remove this earlier).
this line is no longer needed, and can be removed (since CA2007 is already supressed in our Build.targets for test projects)

Comment (oguzhankoral) [reply Automate/Tests/Speckle.Automate.Sdk.Tests.Integration/Speckle.Automate.Sdk.Tests.Integration.csproj]: done and no any following warning because of this

**Review by JR-Morgan — APPROVED** [review]
My comments are resolved,

**Review by AlanRynne — APPROVED** [review]
Approving this! 🚀

File: Automate/Tests/Speckle.Automate.Sdk.Tests.Integration/GetAutomationStatus.cs (hunk 1 above)
Comment (AlanRynne) [reply Automate/Tests/Speckle.Automate.Sdk.Tests.Integration/GetAutomationStatus.cs:36]: Resolving this conversation in favour of the JIRA ticket https://spockle.atlassian.net/browse/CNX-8690
